django/urls/resolvers.py

```diff
@@ -258,7 +259,7 @@ def match(self, path):
                 converter = self.converters[key]
                 try:
                     kwargs[key] = converter.to_python(value)
-                except ValueError:
+                except Http404:
```

Comment:
We shouldn't change the current behavior because it is backward incompatible and it wasn't my intention. IMO we should catch both:
```python
except (Http404, ValueError):
```